### ISSUE TITLE
Add ChildParentDataLink opposite of ParentChildDataLink and and all of its component variants

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1252]
### Is your feature request related to a problem? Please describe.

Right now if you wanna send data from a child to a parent you have to do some work arounds like using dynamic variables are using other methods of doing so but when you're sending data from a parent to a child there's a dedicated component for that so I feel like a opposite component to that would be really fitting

this would be the exact opposite of ParentChildDataLink and its other variants instead of sending data down from a parent this sends data up from a child if there are multiple children the slot order that they're in matters as the top one will be the one that sends the data

### Describe the solution you'd like

I want child variants of the parent link components the data that is sent to the parent will prioritize the first slot in the slot order to take data from

### Describe alternatives you've considered

As described above there are multiple methods to send data from a child to a parent for example using dynamic variables or getting the slot name of the parent from the child then comparing that to a static string the ladder being more fragile than the former

### Additional Context

So this is all within the context of a snapper because I wanna send data from a snappable object onto my avatars snap target but I can't send data from the snapper object to the snap target on my avatar I want the ability to make my eyes squint when my glasses are taken off my avatar I currently have this functionality but it is fragile I originally wanted to use parent data link as I thought that was really suitable for my use case I realized I needed the opposite of it

so I wanted to send a float value from my glasses onto its parent being the snap target so I can then manipulate that value using components to make my eyes squint this would be a more robust way of doing this I was also planning on using a bool variant of it so I could also enable a blur effect

**Comment by JackTheFoxOtter** [comment]
I don't think this would work. Reson being: You can have multiple children on a slot. ParentLink works because no matter which children you are, you always ever have a single parent. But the other way around, when you are a parent and have multiple children, what children should you read the value from?

In those situations I think using dynamic variables are a better option. Or even better, since you're working on a snapper: If there is only one thing that can snap to the snapper, or multiple but you don't care about which one it is, you can just compare the children count of the snapper slot to 0.

**Comment by epicEaston197** [comment]
I already addressed this **the slot order will matter** **_a lot_** so the first child in the slot order would be the one that it gets its value from if you want data from a different child you would simply just edit the slot order so the one that you want data from is at the top

**Comment by JackTheFoxOtter** [comment]
I mean, that would work, but is of questionable utility. Unless you explicitly define it, the slot order is more or less random. This is one of those cases that could potentially become problematic if people don't explicitly specify the slot order.

I still think dynamic variables are your best bet - although you'd still potentially have troubles with values on multiple children overwriting one another.

**Comment by epicEaston197** [comment]
> I mean, that would work, but is of questionable utility. Unless you explicitly define it, the slot order is more or less random. This is one of those cases that could potentially become problematic if people don't explicitly specify the slot order.
> 
> I still think dynamic variables are your best bet - although you'd still potentially have troubles with values on multiple children overwriting one another.

Well your concern is fixed just by having documentation and how the component is supposed to be properly used

**Comment by BaxterOttoman** [comment]
While discussing this in Discord with Easton, I mentioned that when collections are introduced, this would have a lot more utility. Even if it's not practical in the now, I think this component would be a useful one to keep in mind for the future. At that point, you could have the parent add child values to a collection of values and use them however you like.

**Comment by 5H4D0W-X** [comment]
the "child uncertainty" problem could be turned into a feature by providing an enum setting that changes how the values are handled. For example the first child in the hierarchy could be used (useful for snappers and snappable objects) or the values of all children could be averaged, run through a min or max function, added, multiplied and so on

**Comment by JackTheFoxOtter** [comment]
I like the idea of implementing this as a collection feature. That makes much more sense to me.

**Comment by epicEaston197** [comment]
I really like shadows ideal and the thought of collections did not occur to me when making this issue so I suppose my ideal may be too far ahead

**Comment by JackTheFoxOtter** [comment]
If this would be implemented as a collection feature, Shadow's approach would be possible, just more generalized. You'll likely get set operation components for that collection then, like Any / Sum / Average, and could use them in combination. That's much more flexible and generic than hard-coding those in with an enum to cycle through.

**Comment by mpmxyz** [comment]
I've got another workaround to offer instead of dynamic variables:
- The parent offers a field via a parent link.
- The child drives the parent's field by driving the target of its Value-/ReferenceCopy.
This approach may have deficits with multiple children.
It also does not reset after removing the child. (may be desirable)
